### PR TITLE
xmlbuilder: Add more specific types

### PR DESCRIPTION
**Pull request by higuri**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <<https://github.com/oozcitak/xmlbuilder-js/wiki#create>>, <<https://github.com/oozcitak/xmlbuilder-js/wiki#converting-to-string>>, <<https://github.com/oozcitak/xmlbuilder-js/wiki/XML-Prolog>>
- [x] Increase the version number in the header if appropriate.
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

### CODE REVIEW

**Review by wallymathieu — APPROVED** [review]
Looks good! 😄

File: types/xmlbuilder/index.d.ts [hunk 1]
```diff
@@ -67,7 +67,7 @@ declare namespace xmlbuilder {
         importDocument(input: XMLElementOrXMLNode): XMLElementOrXMLNode;
         root(): XMLElementOrXMLNode;
         document(): any;
-        end(options?: Object): string;
+        end(writerOrOptions?: XMLWriterOrXMLToStringOptions): string;
```
Comment (amcasey) [inline types/xmlbuilder/index.d.ts]: The names make me think a union type might have been intended (i.e. do you expect either options or a writer or is it reasonable to specify both?).

Comment (higuri) [reply types/xmlbuilder/index.d.ts]: Latter. It is reasonable for the method to specify both 'XMLWriter' and 'XMLToStringOptions'. But as you thought, my changes did not express it properly.
I changed the name of type. Thank you for your review.

Comment (higuri) [reply types/xmlbuilder/index.d.ts]: Latter. It is reasonable for the method to specify both 'XMLWriter' and 'XMLToStringOptions'. But as you thought, my changes did not express it properly.
I changed the name of type. Thank you for your review.

**Pull request merged**